graphql/src/execution/resolver.rs

```diff
@@ -79,12 +79,15 @@ pub trait Resolver: Clone + Send + Sync {
     /// Resolves a scalar value for a given scalar type.
     fn resolve_scalar_value(
         &self,
+        _parent_object_type: &s::ObjectType,
+        _parent: &BTreeMap<String, q::Value>,
+        _field: &q::Name,
         scalar_type: &s::ScalarType,
         value: Option<&q::Value>,
-    ) -> q::Value {
-        value
+    ) -> Result<q::Value, QueryExecutionError> {
+        Ok(value
             .and_then(|value| value.coerce(scalar_type))
-            .unwrap_or(q::Value::Null)
+            .unwrap_or(q::Value::Null))
```

Comment:
Since the signature now returns a Result, can't we return the error instead of having `unwrap_or`?